R/calculate_study_day.R

```diff
@@ -0,0 +1,142 @@
+#' `calculate_study_day` performs study day calculation
+#' @description
+#' This function takes the an input data.frame and a reference data.frame (which
+#' is DM domain in most cases), and calculate the study day from reference date
+#' and target date. In case of unexpected conditions like reference date is not
+#' unique for each patient, or reference and input dates are not actual dates,
+#' NA will be returned for those records.
+#'
+#' @md
+#' @param sdtm_in input data.frame that contains the target date.
+#' @param dm_domain reference date.frame that contains the reference date.
+#' @param refdt reference date from `dm_domain` that will be used as reference to
+#' calculate the study day.
+#' @param tgdt target date from `sdtm_in` that will be used to calcualte the study
+#' day.
+#' @param study_day_var the new study day variable name in the output. For
+#' example, AESTDY for AE domain for CMSTDY for CM domain.
```

Comment:
```suggestion
#' example, AESTDY for AE domain; CMSTDY for CM domain.
```

Reply:
Thanks, I updated for to and instead of semicolon, see https://github.com/pharmaverse/sdtm.oak/pull/20/commits/0c786855aaeebb682682599132ddd5273754aa66